### PR TITLE
add deprecation warning to resample function

### LINKED ISSUE #139: deprecate legacy resampler?
### Problem

there is a legacy resampler in parsers.  @Sparks29032 was this from before or did you write this then write a better version.   In general, keeping around things that are not used is just technical debt and we should get rid of it. 

### Proposed solution

 If it is something from Pavol days we may want to keep it as other code is likely using it.  If we think the code that is using it should start using the wsinterp instead we should deprectate it which means that when it is called we print a warning message that it is deprecated and will be removed in a version two mid-bumps later but should be replaced with wsinterp, then make issues linked to two bumps later to remove it.

### PR DESCRIPTION
**Pull request by alisnwu**
closes #139

as per discussion in #180 warnings.warn passes pytest under python 3.13, 3.12, 3.11

@sbillinge @bobleesj ready for review

Screenshot 2024-12-07 at 5 16 14 PM

### CODE REVIEW

File: news/resampler-dep-warn.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
+**Added:**
+
+* Deprecation warning for resample function in resampler
```
Comment (sbillinge) [inline news/resampler-dep-warn.rst:3]: we are not adding functionality so move this from here to `Deprecated`

**Review by sbillinge — COMMENTED** [review]
please see inline

**Comment by alisnwu** [comment]
@sbillinge fixed news item, ready for review

File: news/resampler-dep-warn.rst [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+* <news item>
 
 **Changed:**
 
 * <news item>
 
 **Deprecated:**
 
-* <news item>
+* Deprecation warning for resample function in resampler
```
Comment (sbillinge) [inline news/resampler-dep-warn.rst:11]: how about "`resample` function in `resampler`.  Repaced with `wsinterp`" or sthg like that?

This news is user-facing so it should be targeted at someone who might be using the code.

**Review by sbillinge — COMMENTED** [review]
pls see inline.

**Comment by sbillinge** [comment]
@alisnwu why are you closing this?

**Pull request closed without merging**